### DISCUSSION TITLE
Directus should use binary(16) uuids outside of PG

### DISCUSSION THREAD

**Discussion by grefrit** [directus/directus#11076] [Feature Requests]
Hello. Knex 1.0.0 released new feature https://github.com/knex/knex/pull/4836 to support binary uuids outside of PG. I think directus should use them, because char(36) uuids are wasting a lot of space and have slower indexing. It's a bit inconvenient to char(36) as primary keys, so I think that in new projects Directus should bootstrap with binary uuids and it will be awesome to have support for projects that use binary uuids already

**Comment by rijkvanzanten** [comment]
@GrefriT Directus just calls Knex's `.uuid()` method to create those columns, so if knex will now use the correct types consistently, directus will match 👍🏻

**Reply by nickrum** [reply]
Looking at the linked PR, I think we'd have to manually pass the option to use binary UUIDs to `.uuid()` 🤔

**Reply by rijkvanzanten** [reply]
From that PR, it also looks like knex (and the connection libs) will not ensure that the UUID is always converted to/from string-binary, so that further makes things a little tricky (as we're trying to keep the amount of database abstraction work we have to do in Directus itself to a minimum)

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
